src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -592,6 +598,40 @@ void GoogleCalendarSyncAdaptor::finalCleanup()
     if (succeededAccounts.size()) {
         setLastSyncSuccessful(succeededAccounts);
     }
+
+    // Delete any events which are not associated with a notebook.
+    // These events are ghost events, caused by a bug which previously
+    // existed in the sync adapter code due to mkcal deleteNotebook semantics.
```

Comment:
See JB#22313 and JB#26514 and JB#27888 - sadly it seems that this is a case which can occur somehow in released software (I still don't understand precisely how).